RHIDP-7560 - Install and configure plugins from Extensions page (#1281)
redhat-developer/red-hat-developers-documentation-rhdh: modules/dynamic-plugins/con-extensions-managing-plugins.adoc

[id="con-extensions-managing-plugins_{context}"]
= Managing plugins by using Extensions

> You can install and configure plugins by using Extensions.

Reviewer: ```suggestion
You can install and configure plugins by using *Extensions*.
```

[WARNING]

> Installation and configuration of plugins by using Extensions will only work in development environments. This feature is not supported in production environments.

Reviewer: ```suggestion
Installation and configuration of plugins by using *Extensions* will only work in development environments. This feature is not supported in production environments.
```

> In a production environment, users will be notified that plugin installation is not permitted.

Reviewer: ```suggestion
In a production environment, users will be notified that the plugin installation is not permitted.
```

image::rhdh/extensions-restart-plugin-1.png[]

In a development environment: 

* Administrators can install a plugin using the default configuration preloaded in the editor, or modify the configuration before installing. Upon successful installation, users will be notified that a backend restart is required to complete the installation process.

> * When a plugin is installed, administrators can access the Actions drop-down in the plugin’s side panel. Available actions include:

Reviewer: ```suggestion
* When a plugin is installed, administrators can access the *Actions* drop-down list in the side panel of the plugin. Available actions include:
```
(IBM Style Guide suggests to avoid the usage of possessives for inanimate objects)

Reviewer: Should we say "platform engineers" instead of "administrators"?

> * When a plugin is installed, administrators can access the Actions drop-down in the plugin’s side panel. Available actions include:

Reviewer: ```suggestion
* When a plugin is installed, administrators can access the *Actions* drop-down in the side panel of the plugin. Available actions include:
```

** When a plugin is installed, administrators can access the Actions drop-down in the plugin’s side panel. Available actions include:
*** Editing the configuration used during installation
*** Disabling or enabling the plugin
*** After performing any of these actions, users will be notified that a backend restart is required for the changes to take effect.

////
== RBAC permissions
When RBAC permission framework is enabled, for non-admin users to access Extensions UI, the role associated with your user should have the following permission policies associated with it. Add the following in your permission policies configuration file:
[source,yaml]
----
p, role:default/team_a, extensions-plugin, read, allow
p, role:default/team_a, extensions-plugin, create, allow
g, user:default/<login-id/user-name>, role:default/team_a

permission:
      enabled: true
      rbac:
        admin:
          users:
          - name: user:default/gerry-forde
        pluginsWithPermission:
        - catalog
        - scaffolder
        - permission
        - extensions
----
////

// .Procedure
// . Navigate to RBAC.

> // . Click *Create*.

Author: @debsmita1 I've included instructions for using RBAC UI for extensions permissions. Could you review please?

Also for clarity, can plugins be modified by ANY user by using Extensions in a Dev environment without either having RBAC installed or without the necessary permissions?
